pandas/io/sql.py

```diff
@@ -1391,7 +1392,17 @@ def to_sql(
             dtype=dtype,
         )
         table.create()
-        table.insert(chunksize, method=method)
+        try:
+            table.insert(chunksize, method=method)
+        except exc.SQLAlchemyError as err:
+            if (
+                str(err.orig)
```

Comment:
Question - is this a robust/good way to distinguish the error? I can work on doing this differently if not

Reply:
Yeah I think this is an okay way to catch it. You can match via regex to make it a little cleaner